satpy/writers/geotiff.py

```diff
@@ -34,7 +34,12 @@ class GeoTIFFWriter(ImageWriter):
 
         >>> scn.save_datasets(writer='geotiff')
 
-    Un-enhanced float geotiff with NaN for fill values:
+    By default the writer will use the :class:`~satpy.writers.Enhancer` class to
+    linear stretch the data (see :ref:`Enhancements <chapter_enhancements>`).
+    To get Un-enhanced images ``enhance=False`` can be specified which will
+    write a geotiff with the data type of the dataset with NaN for
+    fill values by default. If a geotiff with a certain datatype is desired
```

Comment:
Where is this `NaN` set? It would probably be best to use the `_FillValue` attribute if it is defined and the DataArray is an integer (this is common practice in Satpy) otherwise `NaN`. Obviously, NaN can't be used for integer types so forcing the NaN fill value would cast the data no matter what. If it is possible to not alter the fill value at all that would be the best.

Reply:
I did not change anything regarding the setting of the fill value. This was documented before I just added a little more context to the documentation. So I guess if this was default behaviour in Satpy it should still be.

Reply:
How about editing the last part of that sentence "with NaN for fill values by default" to "with an appropriate fill value". Or you could be clearer about it, but that may be more documentation than is needed.